zebra-state/src/service/finalized_state.rs

```diff
@@ -390,6 +401,10 @@ impl FinalizedState {
                 batch.zs_insert(history_tree_cf, height, history_tree);
             }
 
+            let current_pool = self.current_value_pool();
+            let new_pool = current_pool.update_with_block(block.borrow(), &new_outputs)?;
```

Comment:
Here `new_outputs` seems to be wrong however i am not sure how to get all the utxos with what we have at this stage.

Reply:
@teor2345 can you focus on this specific question and leave everything else away at least by now ?

Currently, with this code we have 3 failing tests with error `provided Outputs (length N) don't have spent OutPoint`:

```
    service::check::tests::utxo::accept_arbitrary_transparent_spend_from_previous_block
    service::check::tests::utxo::reject_duplicate_transparent_spend_in_same_chain_from_previous_block
    service::finalized_state::tests::prop::blocks_with_v5_transactions
```

And also the chain fails to sync pretty early by the same error of not having spend outpoints.

Reply:
You need to get the UTXOs for all the inputs spent in this block:
>     /// `utxos` must contain the [`Utxo`]s of every input in this block,
>     /// including UTXOs created by earlier transactions in this block.
https://github.com/ZcashFoundation/zebra/blob/298ececabe47905b03a08221e91f7c92aaf95cf4/zebra-chain/src/value_balance.rs#L72

Please see my detailed comment here, which links to the code in this method that finds all the UTXOs spent by the block:
https://github.com/ZcashFoundation/zebra/pull/2599/files#r687299353

Reply:
Cool, thanks, that is some improvement, i missed this valuable suggestion in a sea of irrelevant ones.

We still have some tests failing, now they are 2:

```
   service::check::tests::utxo::accept_later_transparent_spend_from_this_block
   service::finalized_state::tests::prop::blocks_with_v5_transactions
```

Reply:
I can see that you're frustrated, and that I'm not helping the way you want.
I'm still working on giving focused feedback. Please be patient.

If there's any specific help you need, please go ahead and ask me.

I'm looking at the tests and this PR now, I should have some suggestions soon.

Reply:
I made a suggestion about `service::check::tests::utxo::accept_later_transparent_spend_from_this_block` near the code that is failing.

I don't see any failures for `service::finalized_state::tests::prop::blocks_with_v5_transactions`, can you copy and paste the logs, or commit the proptest seed as part of this PR?

Reply:
```
failures:

---- service::finalized_state::tests::prop::blocks_with_v5_transactions stdout ----

The application panicked (crashed).
Message:  Utxo for OutPoint should be present
Location: zebra-state/src/service/finalized_state.rs:347

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

The application panicked (crashed).
Message:  Utxo for OutPoint should be present
Location: zebra-state/src/service/finalized_state.rs:347

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

The application panicked (crashed).
Message:  Utxo for OutPoint should be present
Location: zebra-state/src/service/finalized_state.rs:347

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

The application panicked (crashed).
Message:  Utxo for OutPoint should be present
Location: zebra-state/src/service/finalized_state.rs:347

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

The application panicked (crashed).
Message:  Utxo for OutPoint should be present
Location: zebra-state/src/service/finalized_state.rs:347

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
proptest: Aborting shrinking after the PROPTEST_MAX_SHRINK_ITERS environment variable or ProptestConfig.max_shrink_iters iterations (set 4 to a large(r) value to shrink more; current configuration: 4 iterations)

The application panicked (crashed).
Message:  Test failed: Utxo for OutPoint should be present; minimal failing input: (chain, count, network, _history_tree) = (alloc::vec::Vec<zebra_state::request::PreparedBlock><zebra_state::request::PreparedBlock>, len=104, 6, Testnet, HistoryTree(None))
	successes: 0
	local rejects: 0
	global rejects: 0

Location: zebra-state/src/service/finalized_state/tests/prop.rs:20
```

Reply:
This looks like the same error as:
https://github.com/ZcashFoundation/zebra/pull/2599#discussion_r688180679